nav2_planner/src/planner_server.cpp

```diff
@@ -184,25 +184,32 @@ PlannerServer::computePathToPose()
       goal = action_server_->accept_pending_goal();
     }
 
-    RCLCPP_DEBUG(get_logger(), "Attempting to a find path from (%.2f, %.2f) to "
+    RCLCPP_INFO(get_logger(), "Attempting to a find path from (%.2f, %.2f) to "
       "(%.2f, %.2f).", start.pose.position.x, start.pose.position.y,
-      goal->pose.pose.position.x, goal->pose.pose.position.y);
-
-    result->path = planner_->createPlan(start, goal->pose);
+      goal->poses.poses.back().pose.position.x, goal->poses.poses.back().pose.position.y);
+
+    result->path.poses.clear();
+    result->path.poses.push_back(start);
+    for (auto current_waypoint : goal->poses.poses)
```

Comment:
Here we're computing a path from the current location to final path going through all waypoints at each cycle. It might be more efficient to only compute a path to the next one or two waypoints.

Also the dwb controller will be receiving the entire path and given some configurations trying to align with respect to the final pose instead of the next waypoint.